packages/flutter/lib/src/widgets/scroll_activity.dart

```diff
@@ -241,10 +260,16 @@ class ScrollDragController implements Drag {
   @override
   void update(DragUpdateDetails details) {
     assert(details.primaryDelta != null);
+    final bool lastEventWasStationary = _lastDetails is DragUpdateDetails
+        && _lastDetails.primaryDelta == 0.0;
     _lastDetails = details;
     double offset = details.primaryDelta;
-    if (offset == 0.0)
+    if (offset == 0.0) {
+      // No axis motion over 2 consecutive frames breaks previous momentum.
```

Comment:
it's a bit weird that we base this on frames (actually, we base it on user input) rather than on the timestamps. Suppose a platform came along that reported all events in real time, at 1000Hz. Now you lose momentum if you move less than one device pixel per millisecond, which seems a bit drastic, and is a lot more strict than on a device where you only sample the input at 10Hz (e.g. because of high load).

Can we actually use the timestamps somehow? I suppose we could track when was the last update that was non-zero, and cancel the momentum if we're still at zero and the last such update was more than a certain time ago?